packages/block-library/src/gallery/v1/gallery-image.js

```diff
@@ -119,7 +118,7 @@ class GalleryImage extends Component {
 
 		// If a caption text was meanwhile written by the user,
 		// make sure the text is not overwritten by empty captions.
-		if ( caption && ! get( mediaAttributes, [ 'caption' ] ) ) {
+		if ( caption && ! mediaAttributes.caption ) {
```

Comment:
`mediaAttributes` is always an object:

https://github.com/WordPress/gutenberg/blob/efb2fd3623b5b437d37cc9d52fb3f13e6f354161/packages/block-library/src/gallery/shared.js#L16